File: js/src/parser/ast/AssignStmt.js
const ASTNodeTypes = require("./ASTNodeTypes");
const ParseException = require("../util/ParseException");
const {Stmt} = require('./index')
class AssignStmt extends Stmt {
  constructor() {
    super(ASTNodeTypes.ASSIGN_STMT, "assign");
  }
}

const { Factor, Expr } = require("./index");
AssignStmt.parse = it => {
  const stmt = new AssignStmt()
  const tkn = it.peek();
  const factor = Factor.parse(it);
  if (factor == null) {
    throw ParseException.fromToken(tkn);
  }
  stmt.addChild(factor);
  const lexeme = it.nextMatch("=");
  const expr = Expr.parse(it);
  stmt.addChild(expr);
  stmt.setLexeme(lexeme);
  return stmt;
};
module.exports = AssignStmt;

File: TinyScript/java/src/main/java/parser/ast/AssignStmt.java
package parser.ast;

import lexer.TokenType;
import parser.util.ParseException;
import parser.util.PeekTokenIterator;

public class AssignStmt extends Stmt {

    public AssignStmt() {
        super(ASTNodeTypes.ASSIGN_STMT, "assign");
    }
    public static ASTNode parse(PeekTokenIterator it) throws ParseException {
        var stmt = new AssignStmt();
        var tkn = it.peek();
        var factor = Factor.parse(it);
        if(factor == null) {
            throw new ParseException(tkn);
        }
        stmt.addChild(factor);
        var lexeme = it.nextMatch("=");
        var expr = Expr.parse(it);
        stmt.addChild(expr);
        stmt.setLexeme(lexeme);
        return stmt;

    }

}
